src/identity-utils.ts

```diff
@@ -0,0 +1,160 @@
+import { Dictionary } from './utils';
+import { LocalStorageVault } from './vault';
+import Types from './types';
+import { IdentityApiData, UserIdentities } from '@mparticle/web-sdk';
+import { isObject } from './utils';
+import { MParticleWebSDK } from './sdkRuntimeModels';
+
+export interface IKnownIdentities extends UserIdentities {
+    device_application_stamp?: string;
+}
+
+export interface ICachedIdentityCall extends UserIdentities {
+    responseText: string;
+    status: number;
+    expireTimestamp: number;
+}
+
+export const cacheIdentityRequest = (
+    method: string,
+    identities: IKnownIdentities,
+    expireTimestamp: number,
+    idCache: LocalStorageVault<Dictionary>,
+    xhr: XMLHttpRequest
+): void => {
+    let cache: Dictionary<ICachedIdentityCall> = idCache.retrieve() || {};
+    let cacheKey = concatenateIdentities(method, identities);
+
+    cache[cacheKey] = { responseText: xhr.responseText, status: xhr.status, expireTimestamp};
+    idCache.store(cache);
+};
+
+// We need to ensure that identities are concatenated in a deterministic way, so
+// we sort the identities based on their enum.
+// we create an array, set the user identity at the index of the user identity type
+export const concatenateIdentities = (
+    method: string,
+    userIdentities: IKnownIdentities
+): string => {
+    // set DAS first since it is not an official identity type
+    let cacheKey: string = `${method}:device_application_stamp=${userIdentities.device_application_stamp};`;
```

Comment:
This probably can be a convenience method to make things cleaner.

Reply:
I feel like if it's just a function that will return the same thing, that might be over engineering.